src/pages/api/auth/[...nextauth].ts

```diff
@@ -1,30 +1,84 @@
 import NextAuth, { type NextAuthOptions } from 'next-auth'
-import DiscordProvider from 'next-auth/providers/discord'
+import Credentials from 'next-auth/providers/credentials'
+import { compare } from 'bcryptjs'
 
-// Prisma adapter for NextAuth, optional and can be removed
-import { PrismaAdapter } from '@next-auth/prisma-adapter'
 import { prisma } from '../../../server/db/client'
 import { env } from '../../../env/server.mjs'
+import { User } from '@prisma/client'
 
 export const authOptions: NextAuthOptions = {
-  // Include user.id on session
   callbacks: {
-    session({ session, user }) {
-      if (session.user) {
-        session.user.id = user.id
+    // We need this to ensure that the client knows when to log in
+    async session({ session, token }) {
+      if (session && session.user && token && token.sub) {
+        session.user.id = token.sub
       }
       return session
     },
   },
-  // Configure one or more authentication providers
-  adapter: PrismaAdapter(prisma),
+
   providers: [
-    DiscordProvider({
-      clientId: env.DISCORD_CLIENT_ID,
-      clientSecret: env.DISCORD_CLIENT_SECRET,
+    Credentials({
+      name: 'NUS Email',
+      credentials: {
+        email: {
+          label: 'NUS Email',
+          type: 'email',
+          placeholder: 'XXX@u.nus.edu',
+        },
+        password: {
+          label: 'Password',
+          type: 'password',
+        },
+      },
+      authorize: async (credential) => {
+        // Step 1: If no credentials are provided, return null.
+        if (!credential || !credential.email || !credential.password) {
+          return null
+        }
+
+        // Step 2: Destructure and get the email and password.
+        const { email, password } = credential
+
+        try {
+          // Step 3: Get the user by the email
+          const adapterUser = await prisma.user.findUnique({
+            where: { email },
+          })
+
+          if (!adapterUser) return null
+
+          // Step 4: Type cast it to the type of User
+          const account = adapterUser as User
+
+          // If the account is found, challenge the hashPassword with the password
+          const success = await compare(password, account.hashedPassword)
+          if (!success) return null
+
+          // The user object is passed to the session callback
+          return {
+            id: account.id,
+            email: account.email,
+            level: account.level,
+          }
+        } catch (e) {
+          return null
+        }
+      },
     }),
-    // ...add more providers here
   ],
+
+  secret: env.NEXTAUTH_SECRET,
+  session: {
+    strategy: 'jwt',
+    maxAge: 7 * 60 * 60, // <--- The user can only login for 1 hour
```

Comment:
Minor detail but did you mean 7 hours instead of 1 hour in your comment?

Reply:
That is true haha I have changed it